src/redux/reducers/authReducer.js

```diff
@@ -24,11 +25,43 @@ function setTokens(token, refreshToken) {
   }
 }
 
+
+function parseToken(token) {
+  const [, base64Url] = token.split('.');
+  const base64 = base64Url.replace(/-/g, '+').replace(/_/g, '/');
+  const jsonPayload = decodeURIComponent(
+    atob(base64)
+      .split('')
+      .map((c) => {
+        return `%${(`00${c.charCodeAt(0).toString(16)}`).slice(-2)}`;
+      })
+      .join(''),
+  );
+
+  const parsedToken = JSON.parse(jsonPayload);
+
+  const roleKey = Object.keys(parsedToken).find((key) => key.search(('role')) > -1);
+
+  if (roleKey) {
+    const role = parsedToken[roleKey];
+
+    if (Array.isArray(role)) {
+      parsedToken.roles = role;
+    } else {
+      parsedToken.roles = [role];
+    }
+  }
+
+  return parsedToken;
+}
+
 const authReducer = createReducer(initialState, (builder) => {
```

Comment:
Just for info - React Toolkit has a nice feature called `createSlice` - we could consider using it in future

Reply:
I discussed this topic with Przemek before the implementation and both approaches are correct. I just prefer `createReducer`.